include/cuco/operator.hpp

```diff
@@ -62,6 +62,12 @@ struct count_tag {
 struct find_tag {
 } inline constexpr find;  ///< `cuco::find` operator
 
+/**
+ * @brief `for_each` operator tag
+ */
+struct for_each_tag {
+} inline constexpr for_each;  ///< `cuco::for_each` operator
```

Comment:
I see `for_each` as an internal utility as opposed to an actual hash table operator. Need to think more on this.

Reply:
From my standpoint I would treat it as an extension to the STL API that is more suitable for the GPU. Having a "cooperative iterator" instead, which would be closer to the spirit of modern C++ has its drawbacks. For example, how do we ensure users only increment the iterator with the same CG? `for_each` solves this problem by making the probing part internal. We should even be able to redefine any lookup function (`find`, `count`, `retrieve`) that relies on probing with `for_each`, giving us a proper abstraction layer for probing.

Reply:
On a side note I personally find this funtional approach, i.e., "for each found key do X" very appealing. Historic evidence that it is indeed useful comes from warpcore, where many downstream applications (mostly genomics stuff) implemented their custom lookup operations through `for_each` functors.